### DISCUSSION TITLE
Color conversion for heic image does not work

### DISCUSSION THREAD

**Discussion by asuwendi** [ImageMagick/ImageMagick#6716] [Contribute]
### Prerequisites

- [ x ] I have written a descriptive issue title
- [ x ] I have verified that I am using the latest version of ImageMagick
- [ x ] I have searched [open] and [closed] issues to ensure it has not already been reported

### Description
Color conversion for heic image does not work (this works for JPEG). One way to tell is that the target color profile does not get embedded in the output image.

### Steps to Reproduce
1. Sample inputs (a heic image without embedded color profile and an icc color profile):
[files.zip](https://github.com/ImageMagick/ImageMagick/files/2730524/files.zip)
2. ImageMagick command to reproduce:
convert heic-no-embedded-profile.heic -profile P3.icc output-should-have-embedded-p3-profile.heic
3. Check for embedded ICC profile in the output file (in this case no embedded icc profile in the output. In my case, I used exiftool that supports reading heic metadata):
asuwedi-mbp:Downloads Ari$ exiftool output-should-have-embedded-p3-profile.heic | grep -i profile
File Name                       : output-should-have-embedded-p3-profile.heic
asuwedi-mbp:Downloads Ari$

### System Configuration
- ImageMagick version: ImageMagick 7.0.8-22 Q16 x86_64 2018-12-30
- Environment (Operating system, version and so on): macOS Mojave v10.14.2
- Additional information:

**Comment by urban-warrior** [comment]
ImageMagick requires the libheif delegate library to read / write the HEIC image format.  The delegate library supports reading and writing EXIF and XMP profiles, however, currently ImageMagick only supports reading EXIF.  It does not write EXIF, ICC, or XMP profiles.  ImageMagick is open-source.  Consider posting a Git pull request to add this functionality to ImageMagick's HEIC coder, coders/heic.c.

**Comment by asuwendi** [comment]
A quick note to anyone who is trying to fix this: I looked into doing this but in coders/heic.c WriteHEICImage method, there's a color space transformation to YCbCr (status=TransformImageColorspace(image,YCbCrColorspace,exception)) before encoding the heic image. That color space transformation step would also remove the embedded color profile in ImageMagick's Image (since it's now in YCbCr). So fixing this may need a bit more thought here.

**Comment by Zeus64** [comment]
Find below some sample image :

original file:
[picture.zip](https://github.com/ImageMagick/ImageMagick/files/3145302/picture.zip)

heic_converted_to_jpg_with_7_0_7_23_keeping_icc_profile
![heic_converted_to_jpg_with_7_0_7_23_keeping_icc_profile](https://user-images.githubusercontent.com/4325047/57191109-9f187300-6f2a-11e9-97cc-49455546b136.jpg)

heic_converted_to_jpg_with_7_0_8_44_no_icc_profile_found (red color is washed)
![heic_converted_to_jpg_with_7_0_8_44_no_icc_profile_found](https://user-images.githubusercontent.com/4325047/57191114-a8a1db00-6f2a-11e9-8571-6a510ebab1aa.jpg)

heic_converted_to_jpg_with_7_0_7_23_replacing_icc_profile_to_sRGB_profile_via_MagickProfileImage
![heic_converted_to_jpg_with_7_0_7_23_replacing_icc_profile_to_sRGB_profile_via_MagickProfileImage](https://user-images.githubusercontent.com/4325047/57191143-f585b180-6f2a-11e9-85ad-11bac77ea114.jpg)

to convert to sRGB under 7_0_7_23 (7_0_8_44 is not working because of https://github.com/ImageMagick/ImageMagick/issues/1567) the only way I found is to first call MagickTransformImageColorspace(..sRGBColorspace..) and them MagickProfileImage(..sRGBICCProfile...).

**Comment by ikerr** [comment]
I too am encountering colorspace related problems when converting HEIC images and using the `-profile` option.

**Steps to reproduce:**
`convert IMG_6476.HEIC -profile sRGB2014.icc out.jpg`

**Assets:**
https://gist.github.com/ikerr/bbcb4ca22c4b13a8c2850df24e74ea42

**Expected output**: Colors in out.jpg should closely match those in IMG_6476.HEIC
**Actual output**: Colors in out.jpg are "vastly" different from those in IMG_6476.HEIC

![out.jpg](https://gist.githubusercontent.com/ikerr/bbcb4ca22c4b13a8c2850df24e74ea42/raw/ee1b07b4f7bc000117b10daa239d59a1933b16f6/out.jpg)

**ImageMagick version**: Version: ImageMagick 7.0.9-2 Q16 x86_64 2019-10-31
**ImageMagick delegates**: bzlib freetype heic jng jp2 jpeg lcms ltdl lzma openexr png tiff webp xml zlib
**Operating System**: macOS 10.13.6
